### PR TITLE
Introduce error log, increase new site identification time, properly unset onetime jobs for non commerce sites

### LINKED ISSUE #1257: Log WooCommerce page creation
We've noticed that there are sites where the cart/checkout pages are not created (or are created two times)

Before we dig deeper into this, I'd suggest we add logging in the process, to have some data before moving forward with the fix

See more in internal slack thread p1690777603577939-slack-CNA89KY6M

cc @manospsyx @warrendholmes

### PR DESCRIPTION
**Pull request by PanosSynetos**
### Changes proposed in this Pull Request:

Closes #1257 .
- #1257 

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Sync this PR in an existing site
- [ ] Comment out line 47 `'woocommerce_create_pages'                => 'woocommerce_create_pages_callback',`
- [ ] Comment out lines 227-232, to force the job run for older sites
```php
			if ( WCAdminHelper::is_wc_admin_active_for( 60 * MINUTE_IN_SECONDS ) ) {
				update_option( $this->option_prefix . $operation, 'completed', 'no' );
				$this->write_to_log( $operation, 'completed (60 minutes)' );

				return;
			}
```
- [ ] Execute `wp option delete wc_calypso_bridge_one_time_operation_woocommerce_create_pages` - You can access CLI by visiting `/_cli` on a site (when proxied)
- [ ] ~Go to wp-admin and delete pages `shop, cart, checkout, myaccount, refund_returns` and also empty trash~
- [ ] Uncomment line 47 `'woocommerce_create_pages'                => 'woocommerce_create_pages_callback',`
- [ ] Execute `wp option get woocommerce_admin_install_timestamp` - This will trigger the one time job (or go to wp-admin > pages)
- [ ] Get your site's ID - It's the number in parenthesis in `/_cli`
- [ ] Go to logstash https://logstash.a8c.com/logstash/goto/d351e93188b84a27243f627938b86b93 and set the filter to your site id and click refresh

You will see something like

```
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.1449) woocommerce_create_pages: created menu items
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.1187) woocommerce_create_pages: finished WC_Install::create_pages
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.1459) woocommerce_create_pages: COMMIT and cache deleted
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.0065) woocommerce_create_pages: woocommerce_page_created action - id: 1406, slug: refund_returns
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.0517) woocommerce_create_pages: initialized
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1442) woocommerce_create_pages: deleted WooCommerce page refund_returns
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3303) woocommerce_create_pages: deleted WooCommerce page id for page checkout
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3893) woocommerce_create_pages: woocommerce_create_pages filter - pages:shop, cart, checkout, myaccount, refund_returns
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.0778) woocommerce_create_pages: deleted WooCommerce page shop
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1392) woocommerce_create_pages: deleted WooCommerce page checkout
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.2186) woocommerce_create_pages: deleted WooCommerce page id for page cart
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.4261) woocommerce_create_pages: woocommerce_page_created action - id: 1398, slug: shop
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.098) woocommerce_create_pages: deleted WooCommerce page cart
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.2744) woocommerce_create_pages: deleted WooCommerce page id for page myaccount
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.0519) woocommerce_create_pages: START TRANSACTION
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.5792) woocommerce_create_pages: woocommerce_page_created action - id: 1400, slug: cart
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1602) woocommerce_create_pages: deleted WooCommerce page id for page shop
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3878) woocommerce_create_pages: set cart/checkout blocks
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.729) woocommerce_create_pages: woocommerce_page_created action - id: 1402, slug: checkout
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1189) woocommerce_create_pages: deleted WooCommerce page my-account
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.8789) woocommerce_create_pages: woocommerce_page_created action - id: 1404, slug: my-account
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3864) woocommerce_create_pages: deleted WooCommerce page id for page refund_returns
```

If you notice things happening out of order, you're right; you have a keen eye for details.

Logstash does not keep the time with milliseconds. I've introduced the time in milliseconds in parenthesis, so we can sort externally in an editor and see the events in the order they happened.

Here's a sorted version:
```
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.0517) woocommerce_create_pages: initialized
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.0519) woocommerce_create_pages: START TRANSACTION
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.0778) woocommerce_create_pages: deleted WooCommerce page shop
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.098) woocommerce_create_pages: deleted WooCommerce page cart
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1189) woocommerce_create_pages: deleted WooCommerce page my-account
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1392) woocommerce_create_pages: deleted WooCommerce page checkout
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1442) woocommerce_create_pages: deleted WooCommerce page refund_returns
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.1602) woocommerce_create_pages: deleted WooCommerce page id for page shop
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.2186) woocommerce_create_pages: deleted WooCommerce page id for page cart
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.2744) woocommerce_create_pages: deleted WooCommerce page id for page myaccount
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3303) woocommerce_create_pages: deleted WooCommerce page id for page checkout
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3864) woocommerce_create_pages: deleted WooCommerce page id for page refund_returns
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3878) woocommerce_create_pages: set cart/checkout blocks
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.3893) woocommerce_create_pages: woocommerce_create_pages filter - pages:shop, cart, checkout, myaccount, refund_returns
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.4261) woocommerce_create_pages: woocommerce_page_created action - id: 1398, slug: shop
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.5792) woocommerce_create_pages: woocommerce_page_created action - id: 1400, slug: cart
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.729) woocommerce_create_pages: woocommerce_page_created action - id: 1402, slug: checkout
Aug 4, 2023 @ 09:33:59.000	WooExpress: Operation: (1691141639.8789) woocommerce_create_pages: woocommerce_page_created action - id: 1404, slug: my-account
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.0065) woocommerce_create_pages: woocommerce_page_created action - id: 1406, slug: refund_returns
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.1187) woocommerce_create_pages: finished WC_Install::create_pages
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.1449) woocommerce_create_pages: created menu items
Aug 4, 2023 @ 09:34:00.000	WooExpress: Operation: (1691141640.1459) woocommerce_create_pages: COMMIT and cache deleted
```

------------------

This PR also increases the old site identification, from 5 minutes, to one hour.
We've noticed that 5 minutes is too short, and might be possible to cause the pages not being created and the job marked as complete (when it's not)

-------------------

I've also found that I was not unsetting the one-time job keys properly, for non commerce sites. Accidentally I was unsetting using the callback name, instead of the key. No need to test this, it's obvious :)

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: includes/class-wc-calypso-bridge-setup.php [hunk 1]
```diff
@@ -258,6 +270,7 @@ public function woocommerce_create_pages_callback() {
 
 			if ( 'completed' === $status ) {
 				$wpdb->query( 'ROLLBACK' );
+				$this->write_to_log( $operation, 'ROLLBACK' );
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-setup.php:273]: How about adding a "already completed"

File: includes/class-wc-calypso-bridge-setup.php [hunk 2]
```diff
@@ -60,6 +60,14 @@ public static function get_instance() {
 	 */
 	protected $option_prefix = 'wc_calypso_bridge_one_time_operation_';
 
+	/**
+	 * Log prefix.
+	 *
+	 * @since xxx
+	 * @var string
+	 */
+	protected $log_prefix = 'WooExpress: Operation: ';
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-setup.php:69]: Let's omit this prop. It's only used in the `write_log` helper.

File: includes/class-wc-calypso-bridge-setup.php [hunk 3]
```diff
@@ -138,10 +146,10 @@ public function modify_one_time_operations() {
 		if ( ! wc_calypso_bridge_has_ecommerce_features() ) {
 			unset( $this->one_time_operations[ 'set_jetpack_defaults' ] );
 			unset( $this->one_time_operations[ 'woocommerce_create_pages' ] );
-			unset( $this->one_time_operations[ 'set_wc_tracker_twice_daily_callback' ] );
-			unset( $this->one_time_operations[ 'set_wc_tracker_default_callback' ] );
-			unset( $this->one_time_operations[ 'set_wc_subscriptions_siteurl_callback' ] );
-			unset( $this->one_time_operations[ 'set_wc_subscriptions_siteurl_add_domain_callback' ] );
+			unset( $this->one_time_operations[ 'set_wc_tracker_twice_daily' ] );
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-setup.php:149]: Nice catch! 👌

File: includes/class-wc-calypso-bridge-setup.php [hunk 4]
```diff
@@ -273,6 +286,7 @@ public function woocommerce_create_pages_callback() {
 					$page = get_page_by_path( $page, ARRAY_A );
 					if ( is_array( $page ) && isset( $page['ID'] ) ) {
 						wp_delete_post( $page['ID'], true );
+						$this->write_to_log( $operation, 'deleted WooCommerce page ' . $page );
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-setup.php:289]: The `$pages` is probably an array. We should use the page name

**Review by xristos3490 — APPROVED** [review]
Looks good! I pointed some few minor things and we are good to go! 

Pre-Approving.

File: includes/class-wc-calypso-bridge-setup.php (hunk 4 above)
Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-setup.php:289]: This `$page` is not an array, it's in a loop of an array

Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-setup.php:289]: Scratch that, good catch

**Review by ilyasfoo — APPROVED** [review]
Tested well! I've updated changelog and versions in https://github.com/Automattic/wc-calypso-bridge/pull/1258/commits/84abcd7a81b0b1d4abe21ede288367eb7434fd36. Will merge after CI is green.

**Pull request merged**